cpp/src/unary/cast_ops.cu

```diff
@@ -137,7 +140,8 @@ template <typename From, typename To>
 constexpr inline auto is_supported_fixed_point_cast()
 {
   return (cudf::is_fixed_point<From>() && cudf::is_numeric<To>()) ||
-         (cudf::is_numeric<From>() && cudf::is_fixed_point<To>());
+         (cudf::is_numeric<From>() && cudf::is_fixed_point<To>()) ||
+         (cudf::is_fixed_point<From>() && cudf::is_fixed_point<To>());
```

Comment:
This implies decimal types are not numeric (is_numeric seems to indicate this).  That seems incorrect.

Reply:
@harrism has brought this up previously, but this is intentional. Although intuitively it makes sense for `fixed_point` to be next to `floating_point` in `is_numeric`, `fixed_point` is **fundamentally** different than numeric types because we store part of the type in `data_type`.